test/e2e/subscription_e2e_test.go

```diff
@@ -2813,48 +2816,58 @@ func checkDeploymentWithPodConfiguration(t GinkgoTInterface, client operatorclie
 		return
 	}
 
-	check := func(container *corev1.Container) {
+	check := func(container *corev1.Container) error {
 		for _, e := range envVar {
 			existing, found := findEnvVar(container.Env, e.Name)
-			require.Truef(t, found, "env variable name=%s not injected", e.Name)
-			require.NotNil(t, existing)
+			if !found || existing == nil {
+				return errors.New(fmt.Sprintf("env variable name=%s not injected", e.Name))
+			}
 			require.Equalf(t, e.Value, existing.Value, "env variable value does not match %s=%s", e.Name, e.Value)
```

Comment:
Mentioned above: if we're going to be working towards error handling in this inline check function, then I think we should avoid using any testify assertions, and instead returning an error if the conditional check had failed. WYDT?

Reply:
I agree.  Let me try.